src/backend/WorkflowManagement/WorkflowManager.cs

```diff
@@ -0,0 +1,75 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System.Threading.Tasks;
+
+namespace WorkflowManagement
+{
+    public enum WorkflowStage
+    {
+        Analyze,
+        Report,
+        Telemetry,
+        Finished
+    }
+
+    /// <summary>
+    /// Workflow stages are in the following order: Analyze -> Report -> Telemetry -> Finished
+    /// </summary>
+    public class WorkflowManager
+    {
+        readonly IWorkflowAction[] actions;
+        static WorkflowManager manager;
+
+        public WorkflowManager()
+        {
+            IWorkflowAction[] workflowActions = new IWorkflowAction[3];
+
+            workflowActions[(int)WorkflowStage.Analyze] = new AnalyzeAction();
```

Comment:
Instead of hard-coding these, it might be better to have `IWorkflowAction` implementations report which stages they handle (through a method or property added to the interface) so that we can add new workflow actions (which will eventually be made available through DI, hopefully) without having to update WorkflowManager logic. Just tell it to invoke the first workflow manager implementation that can handle the stage we're on.

Reply:
good suggestion - I have made both this change and the next one.